### DISCUSSION TITLE
File Location - CDE 11556141

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#57] [API Design Discussion]
caDSR Link: [11556141](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=11556141%20and%20ver_nr=1)

Definition: The specification of a node (file or directory) in a hierarchical file system where an electronic file is located.

Discussion: From the definition this would appear to be a file path (ex. /Users/aglyle/Desktop/SAMPLE_INFO.txt).  Is this a useful field for helping users discover our data?  Will users primarily have to request access to files and then receive them through a data portal?  Would a hyperlink address be useful?

**Comment by allisonheath** [comment]
For Kids First, I was thinking this location would actually be our GA4GH DRS URL, e.g. drs://data.kidsfirstdrc.org/1ceb95b7-0056-4b76-b6a6-4d586cabb739. Per the spec this means that further standardized metadata around the file is available at https://data.kidsfirstdrc.org/ga4gh/drs/v1/objects/1ceb95b7-0056-4b76-b6a6-4d586cabb739 (which then has the s3 URL, checksums and file names). 

This is what we use for NCPI and being able to bring data files into both CAVATICA and Terra. It's a fairly easy standard to setup and have been working in CFDE with other DCCs who have set it up as well (dbGaP also supports DRS and I believe CDS as well?).

**Comment by Ssandor13** [comment]
Our St. Jude team discussed a few things about this CDE and would like to hear further from the Federation, if possible.
1. From an end user experience, having a file location may not make sense since, specifically, our locations are dynamic and we would not be able to expose/share those. Instead, the end user likely would like an aggregated sample or subject URL so they can go request the files for those samples/subjects rather than trying to identify which file goes with which sample/subject in each data commons.
2. File locations should likely not be related to the metadata and should instead only be communicated through special facilities (gateways and links) in the API. What do others think about this?

**Comment by GeoffLyle** [comment]
Current use for this CDE is as a S3 bucket path, which is then converted to a DRS UUID.  Goal of CDE is to enable users to access files.

Difficulties arise when the URL to a file location is dynamic or is controlled access.  For CHOP, if a user tries to access a controlled access file, there will be an error and their request will not go through.  Treehouse is currently sharing only open access files, will need to discuss how to handle controlled access files.  St. Jude has issue with being unable to expose dynamic file locations, could this URL point to data platform portal where users could request access.

**Comment by claymcleod** [comment]
From my perspective, this should concept should not be a harmonized metadata element, as I don't consider it to be "metadata" as we've been defining it under this API. Instead, this is more akin to actual data, and it should be covered by the `Link` and `Gateway` concepts.

**Comment by lydiabrogers** [comment]
Harmonization Team meeting 2/16 - CDE complete not required.
